src/app/src/components/AnimatedMap.tsx

```diff
@@ -18,41 +20,50 @@ import StatesLayer from './StatesLayer';
 import TimeControlIcon from './TimeControlIcon';
 
 import {
-    MonthlySpendingOverTimeResponse,
+    MonthlySpendingOverTimeByState,
     SpendingByGeographyAtMonth,
 } from '../types/api';
-import { spendingDataByMonth } from './dummySpendingDataByMonth';
 import AnimatedMapLegend from './AnimatedMapLegend';
+import { useGetSpendingOverTimeQuery } from '../api';
 
-const PROGRESS_FINAL_MONTH = 26;
+const START_YEAR = 2021;
+const END_DATE = new Date();
+const PROGRESS_FINAL_STEP = () => {
+    const final_month_step = END_DATE.getMonth();
+    const final_year_step = END_DATE.getFullYear();
+    return 12 * (final_year_step - START_YEAR) + final_month_step;
+};
```

Comment:
Nice idea to encapsulate this logic in a closure. Unfortunately, it's re-calculated every time it's used. You can run it just once by calling the function here:
```suggestion
const PROGRESS_FINAL_STEP = (() => {
    const final_month_step = END_DATE.getMonth();
    const final_year_step = END_DATE.getFullYear();
    return 12 * (final_year_step - START_YEAR) + final_month_step;
})();
```
Then you can remove the parens from the later references to `PROGRESS_FINAL_STEP`.

Reply:
This is cool! Didn't realize I could do that, added that fix as part of 6149ddb5ae5ff20fb7838ba2e051df6fb521995f